### PR TITLE
Add binutils variant +plugins

### PR DESCRIPTION
**Pull request by healther**

### CODE REVIEW

File: var/spack/repos/builtin/packages/binutils/package.py [hunk 1]
```diff
@@ -46,6 +46,7 @@ class Binutils(Package):
     # OpenSpeedShop and cbtf-krell
     variant('krellpatch', default=False,
             description="build with openspeedshop based patch.")
+    variant('plugins', default=False)
```
Comment (davydden) [inline var/spack/repos/builtin/packages/binutils/package.py]: description would be good

Comment (healther) [inline var/spack/repos/builtin/packages/binutils/package.py]: Sorry I don't have one, it is needed to build the gold linker c.f. https://github.com/unsw-corg/SVF/wiki/Install-LLVM-Gold-Plugin-on-Ubuntu

Comment (davydden) [inline var/spack/repos/builtin/packages/binutils/package.py]: > Sorry I don't have one

you don't have to have it, just add one :smile:
i.e. `enable plugins needed for gold linker`.

**Comment by healther** [comment]
This was a reaction to #2554 and pinning to <2.6 was the simplest solution

File: var/spack/repos/builtin/packages/binutils/package.py (hunk 1 above)
Comment (healther) [inline var/spack/repos/builtin/packages/binutils/package.py]: done

**Pull request merged**